src/bindings/python/src/pyopenvino/graph/symbol.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "openvino/core/symbol.hpp"  // ov::Symbol
+
+#include <pybind11/pybind11.h>
+
+#include "pyopenvino/graph/symbol.hpp"
+
+namespace py = pybind11;
+
+void regclass_graph_Symbol(py::module m) {
+    py::class_<ov::Symbol, std::shared_ptr<ov::Symbol>> symbol(m, "Symbol");
+    symbol.doc() = "openvino.runtime.Symbol wraps ov::Symbol";
+
+    symbol.def(py::init([]() {
+        return std::make_shared<ov::Symbol>();
+    }));
+
+    symbol.def(
+        "__eq__",
+        [](const std::shared_ptr<ov::Symbol>& a, const std::shared_ptr<ov::Symbol>& b) {
+            return ov::symbol::are_equal(a, b);
+        },
+        py::is_operator());
```

Comment:
How about other function like `ancestor_of`? Are there any reasons to not include them? https://github.com/openvinotoolkit/openvino/blob/c3c409ee133ffb26bf8fd5570ef50a7c004839a4/src/core/include/openvino/core/symbol.hpp#L15-L22

Reply:
These are in the development mode, so we'd like to secure python API from frequent changes.
Additionally, user scenarios that we can think of don't require these functions.